src/compiler/scala/tools/nsc/transform/UnCurry.scala

```diff
@@ -775,7 +775,7 @@ abstract class UnCurry extends InfoTransform
                       tpe
                   }
                 val info = info0.normalize
-                val tempValName = unit freshTermName (p.name + "$")
+                val tempValName = unit.freshTermName(s"${p.name.decoded}$$")
```

Comment:
Why `.decoded` here?

Reply:
I got finicky because distracted trying to remember how to make a dollar sign. I'll revert.

My automatic rewrite is to avoid toString.

Reply:
I expected it to bomb:
```
    def f(c: $line3.iw$C, +: $line3.iw$C$D): String = {
      <synthetic> val +$1: $line3.iw$C$D = +;
      +$1.toString()
    };
```

Reply:
PR #8994